compiler/passes/insertWideReferences.cpp

```diff
@@ -1895,14 +1901,33 @@ static void insertWideTemp(Type* type, SymExpr* src) {
   Expr* rhs = src->copy();
   if (needsAddrOf)
     rhs = new CallExpr(PRIM_ADDR_OF, rhs);
-  if (needsDeref) {
+  else if (needsDeref) {
```

Comment:
Applause.  I like that it's clearer that needsAddrOf and needDeref can't be simultaneously true.
As a footnote this makes it seem as if an enum might be a good choice one day if this logic becomes
more confusing.